### PR TITLE
Add file decompression function when not in Windows system

### PR DESCRIPTION
**Pull request by mannaandpoem**
**Features**
Add file decompression function when not in Windows system

### CODE REVIEW

File: tests/metagpt/test_incremental_dev.py [hunk 1]
```diff
@@ -187,3 +191,5 @@ def check_or_create_base_tag(project_path):
 
 if __name__ == "__main__":
     pytest.main([__file__, "-s"])
+
+"""pytest -s tests/metagpt/test_incremental_dev.py --doctest-modules --cov=./metagpt --cov-report=html:htmlcov --durations=20"""
```
Comment (garylin2099) [inline tests/metagpt/test_incremental_dev.py:195]: 这一行不用加

Comment (mannaandpoem) [reply tests/metagpt/test_incremental_dev.py]: 好的👌

**Comment by geekan** [comment]
@garylin2099 mark this pr as approve (in files tab) if it's ok

File: metagpt/actions/summarize_code.py [hunk 2]
```diff
@@ -109,8 +109,8 @@ async def run(self):
             code_blocks.append(code_block)
         format_example = FORMAT_EXAMPLE
         prompt = PROMPT_TEMPLATE.format(
-            system_design=design_doc.content,
-            task=task_doc.content,
+            system_design=design_doc.content if design_doc else "",
```
Comment (iorisa) [inline metagpt/actions/summarize_code.py:112]: System design provides complete information on class division, data flow, and control flow. 
Without this, there is no global perspective. Tasks include information on the division of files, and this is also indispensable.

Comment (iorisa) [reply metagpt/actions/summarize_code.py:112]: 如果不知道system design的文件名，可以通过搜task.json里的task list来判断。
如果条件不具备，就不用执行这个action了。

Comment (mannaandpoem) [reply metagpt/actions/summarize_code.py:112]: OK，这一块改回去了。之前找不到文件名是因为在获取dependencies时，windows的 `\\` 与 `/` 不匹配的原因

**Pull request closed without merging**